test/mocha/20-recovery.js

```diff
@@ -102,22 +102,27 @@ describe('Failure recovery', function() {
     credential.id = 'urn:someId3';
     const {token} = secrets;
 
-    await api.post(
+    const result1 = await api.post(
       '/issue',
       {credential},
       {headers: {Authorization: `Bearer ${token}`}}
     );
-
-    const result = await api.post(
+    // expect a duplicate error when the same credential is posted a second time
+    const result2 = await api.post(
       '/issue',
       {credential},
       {headers: {Authorization: `Bearer ${token}`}}
     );
-    result.status.should.equal(409);
-    should.exist(result.data);
-    result.data.should.be.an('object');
-    result.data.message.should.equal('Could not issue credential; duplicate ' +
+    result1.status.should.equal(200);
```

Comment:
move assertions about `result1` up to immediately after the request is made.  This will mean if something goes wrong on the first request, the assertion will fail at that point and the test is aborted.